iis-wf/iis-wf-report/src/main/java/eu/dnetlib/iis/wf/report/pushgateway/converter/GaugesBuilder.java

```diff
@@ -0,0 +1,56 @@
+package eu.dnetlib.iis.wf.report.pushgateway.converter;
+
+import eu.dnetlib.iis.common.java.stream.ListUtils;
+import io.prometheus.client.Gauge;
+
+import java.util.List;
+
+public class GaugesBuilder {
+
+    private GaugesBuilder() {
+    }
+
+    @FunctionalInterface
+    public interface BuilderWithLabels {
+        Gauge build(String metricName,
+                    String help,
+                    List<String> labelNames,
+                    List<List<String>> labelValues,
+                    List<Double> values);
+    }
+
+    public static Gauge buildGaugeWithLabels(String metricName,
+                                             String help,
+                                             List<String> labelNames,
+                                             List<List<String>> labelValues,
+                                             List<Double> values) {
+        Gauge gauge = Gauge.build()
+                .name(metricName)
+                .help(help)
+                .labelNames(labelNames.toArray(new String[0]))
```

Comment:
I guess it will be optimal to initialize the `String` array with the known size (`labelNames#size()`).

Reply:
I did that, but IDE flagged this and suggested to convert it to the version with `new String[0]`. Here's the description IDE gives:

Call to 'toArray()' with pre-sized array argument 'new String[labelNames.size()]' 
 Inspection info: There are two styles to convert a collection to an array: either using a pre-sized array (like c.toArray(new String[c.size()])) or using an empty array (like c.toArray(new String[0]).
In older Java versions using pre-sized array was recommended, as the reflection call which is necessary to create an array of proper size was quite slow. However since late updates of OpenJDK 6 this call was intrinsified, making the performance of the empty array version the same and sometimes even better, compared to the pre-sized version. Also passing pre-sized array is dangerous for a concurrent or synchronized collection as a data race is possible between the size and toArray call which may result in extra nulls at the end of the array, if the collection was concurrently shrunk during the operation.
This inspection allows to follow the uniform style: either using an empty array (which is recommended in modern Java) or using a pre-sized array (which might be faster in older Java versions or non-HotSpot based JVMs).